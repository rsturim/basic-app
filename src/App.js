import React from 'react';

import { BrowserRouter as Router, Route } from 'react-router-dom';

import ComponentOne from './pages/ComponentOne';
import Address from './pages/Address';
import Signup from './pages/Signup';
import Planets from './pages/Planets';

const App = () => (
    <Router>
        <div>
            <Route exact path="/" component={ComponentOne} />
            <Route path="/one" component={ComponentOne} />
            <Route path="/address" component={Address} />
            <Route path="/signup" component={Signup} />
            <Route path="/planets" component={Planets} />
        </div>
    </Router>
);

export default App;
