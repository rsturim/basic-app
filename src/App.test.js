import React from 'react';
import App from './App';

import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

describe('App', () => {
    test('should find and verify and item in the DOM', () => {
        const wrapper = mount(<App />);

        expect(wrapper.find('ComponentOne').length).toEqual(1);
        expect(wrapper.find('.nav-pills li').length).toEqual(4);
    });

    test('should match the snapshot', () => {
        const wrapper = mount(<App />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
