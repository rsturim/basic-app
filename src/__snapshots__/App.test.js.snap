// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<App>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        >
          <Planets
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div
              className="container"
            >
              <Header>
                <h3
                  className="display-5"
                >
                  Planet List
                </h3>
              </Header>
              <Nav
                item="/planets"
              >
                <div>
                  <ul
                    className="nav nav-pills mt-4 mb-4"
                  >
                    <li
                      className="nav-item"
                      key="One"
                    >
                      <Link
                        className="nav-link"
                        to="/one"
                      >
                        <a
                          className="nav-link"
                          href="/one"
                          onClick={[Function]}
                        >
                          One
                        </a>
                      </Link>
                    </li>
                    <li
                      className="nav-item"
                      key="Two"
                    >
                      <Link
                        className="nav-link"
                        to="/two"
                      >
                        <a
                          className="nav-link"
                          href="/two"
                          onClick={[Function]}
                        >
                          Two
                        </a>
                      </Link>
                    </li>
                    <li
                      className="nav-item"
                      key="Sign Up"
                    >
                      <Link
                        className="nav-link"
                        to="/signup"
                      >
                        <a
                          className="nav-link"
                          href="/signup"
                          onClick={[Function]}
                        >
                          Sign Up
                        </a>
                      </Link>
                    </li>
                    <li
                      className="nav-item"
                      key="Planets"
                    >
                      <a
                        className="nav-link active"
                        href="#"
                      >
                        Planets
                      </a>
                    </li>
                  </ul>
                </div>
              </Nav>
              <PlanetList />
            </div>
          </Planets>
        </Route>
        <Route
          component={[Function]}
          path="/one"
        />
        <Route
          component={[Function]}
          path="/two"
        />
        <Route
          component={[Function]}
          path="/signup"
        />
        <Route
          component={[Function]}
          path="/planets"
        />
      </div>
    </Router>
  </BrowserRouter>
</App>
`;
